mono/mini/mini.c

```diff
@@ -3880,20 +3881,25 @@ mini_method_compile (MonoMethod *method, guint32 opts, MonoDomain *domain, JitFl
 	mono_perfcounters->jit_methods++;
 	mono_perfcounters->jit_bytes += header->code_size;
 #endif
-	mono_jit_stats.allocated_code_size += cfg->code_len;
-	code_size_ratio = cfg->code_len;
-	if (code_size_ratio > mono_jit_stats.biggest_method_size && mono_jit_stats.enabled) {
-		mono_jit_stats.biggest_method_size = code_size_ratio;
-		g_free (mono_jit_stats.biggest_method);
-		mono_jit_stats.biggest_method = g_strdup_printf ("%s::%s)", method->klass->name, method->name);
-	}
-	code_size_ratio = (code_size_ratio * 100) / header->code_size;
-	if (code_size_ratio > mono_jit_stats.max_code_size_ratio && mono_jit_stats.enabled) {
-		mono_jit_stats.max_code_size_ratio = code_size_ratio;
-		g_free (mono_jit_stats.max_ratio_method);
-		mono_jit_stats.max_ratio_method = g_strdup_printf ("%s::%s)", method->klass->name, method->name);
+	/* FIXME: use an explicit function to read booleans */
+	if ((gboolean)InterlockedRead ((gint32*)&mono_jit_stats.enabled)) {
+		InterlockedAdd (&mono_jit_stats.allocated_code_size, cfg->code_len);
```

Comment:
isn't that a semantic change? now `allocated_code_size` gets increased only if `mono_jit_stats.enabled` is set. before it was unconditional.

Reply:
You're right of course, that slipped through! Technically, I cannot think of a good reason to increase the counter while the stats are disabled but I will revert that, just in case ... lots of interesting stuff happened already while cleaning up the counters; no need to provoke the Mono spirits! :D